### PR TITLE
Add e2e test prototype for Cluster lifecycle API

### PR DESCRIPTION
**Pull request by g-gaston**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/executables/kubectl.go [hunk 1]
```diff
@@ -210,14 +210,9 @@ func (k *Kubectl) LoadSecret(ctx context.Context, secretObject string, secretObj
...
 
-func (k *Kubectl) ApplyKubeSpec(ctx context.Context, cluster *types.Cluster, spec string) error {
-	params := []string{"apply", "-f", spec}
-	if cluster.KubeconfigFile != "" {
-		params = append(params, "--kubeconfig", cluster.KubeconfigFile)
-	}
-	_, err := k.Execute(ctx, params...)
-	if err != nil {
-		return fmt.Errorf("executing apply: %v", err)
+func (k *Kubectl) ApplyManifest(ctx context.Context, kubeconfigPath, manifestPath string) error {
```
Comment (mitalipaygude) [inline pkg/executables/kubectl.go:213]: nit: why the change from `spec` to `manifestPath`?

Comment (g-gaston) [reply pkg/executables/kubectl.go:213]: spec is a bit ambiguous IMO
What is spec? the spec of an object? the spec of a cluster?
what this argument is is the path to a file containing a yaml manifest

**Review by cxbrowne1207 — COMMENTED** [review]
/lgtm
